### PR TITLE
Add task errors to UI backend, fix config issue

### LINKED ISSUE #180: Show connector status
If a connector or task is in FAILED state, we'll need at least to have a hover or something with the connector's error message.

@rk3rn3r, I suppose this may require some new backend method, too.

### PR DESCRIPTION
**Pull request by rk3rn3r**
+ add task errors to UI backend
+ add task errors to UI frontend tooltips
* fix config issue in prod / UI-backend mode

closes #180

### CODE REVIEW

**Comment by rk3rn3r** [comment]
FYI @uidoyen

File: ui/packages/services/src/config/config.service.ts [hunk 1]
```diff
@@ -17,27 +17,35 @@
...
-            this.config.artifacts.url = w.UI_BASE_URI;
-            console.info("[ConfigService] Applied UI_BASE_URI (" + this.config.artifacts.url + ")!");
-        }
-        if (w.UI_MODE) {
-            this.config.mode = w.UI_MODE;
-            console.info("[ConfigService] Applied UI_MODE (" + this.config.mode + ")!");
+        if (w.UI_CONFIG) {
+            this.config = _.extend({}, this.config, w.UI_CONFIG);
+            console.info("[ConfigService] Applied UI_CONFIG:");
+            console.info(w.UI_CONFIG);
```
Comment (rk3rn3r) [inline ui/packages/services/src/config/config.service.ts:48]: this and above is the fix for the config issue @uidoyen. Pls have a look

**Comment by rk3rn3r** [comment]
@mdrillin @uidoyen @indraraj pls have a look at this. this fixes the config issue in prod/ui-backend mode. `config.js` is served by a backend endpoint in prod/ui-backend mode and you can put overrides for the default config in `config.js`. I also added the backend code to send also the task error messages to the FE. I also added it to the tooltip already. Maybe you would like to make the code nicer and style the content a bit?

@gunnarmorling you might want to review the Java part?

File: ui/config/config.js [hunk 2]
```diff
@@ -1,14 +1,4 @@
...
-    ui: {
-        contextPath: null,
-        url: "http://localhost:8888/ui"
-    },
-    features: {
-        readOnly: false
-    }
+// put UI config overrides here (see config.service.ts)
+window.UI_CONFIG = {
+
```
Comment (rk3rn3r) [inline ui/config/config.js:3]: @uidoyen like we agreed, as this is only for overrides from defaults we make it empty for now.

**Review by indraraj — CHANGES_REQUESTED** [review]

File: ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx [hunk 3]
```diff
@@ -30,13 +30,22 @@ export const ConnectorTask: React.FunctionComponent<IConnectorTaskProps> = (
       color = "grey";
       break;
   }
+  if (!props.errors) {
```
[the remark this answers is not in the document]
Comment (rk3rn3r) [reply ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx:33]: it solves my problem, I just did not want to break the FE b y changing the BE now. Feel free to fix my mistakes!

**Comment by gunnarmorling** [comment]
@rk3rn3r, no comments really on the Java part, LGTM. Btw. is the Delete button working by now? Seems I should record another demo next week then, so to share the latest one publicly.

File: ui/config/config.js (hunk 2 above)
Comment (uidoyen) [reply ui/config/config.js:3]: yeah looks good.

File: ui/packages/services/src/config/config.service.ts (hunk 1 above)
Comment (uidoyen) [reply ui/packages/services/src/config/config.service.ts:48]: Looks good.

File: ui/packages/ui/src/app/pages/connectors/ConnectorListItem.tsx [hunk 4]
```diff
@@ -23,8 +23,9 @@ import { ConnectorTask } from './ConnectorTask';
 export interface IConnectorListItemProps {
   name: string;
   status: string;
-  taskStates: any,
+  taskStates: any;
   type: string;
+  errors: any;
```
Comment (indraraj) [inline ui/packages/ui/src/app/pages/connectors/ConnectorListItem.tsx:28]:
```suggestion
```
not needed here

File: ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx [hunk 5]
```diff
@@ -31,12 +32,23 @@ export const ConnectorTask: React.FunctionComponent<IConnectorTaskProps> = (
...
+    errors =  props.errors;
+  }
+
   return (
     <>
       <Tooltip
         content={
           <div>
             Task {props.taskId}: {props.task}
+            <div class="errors">
```
Comment (indraraj) [inline ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx:46]:
```suggestion
            <div className="errors">
```

File: ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx [hunk 6]
```diff
@@ -31,12 +32,23 @@ export const ConnectorTask: React.FunctionComponent<IConnectorTaskProps> = (
...
+              {errors.map((errormsg, key) => {
+                  return <div class="error" key={key}>{errormsg}</div>;
```
Comment (indraraj) [inline ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx:48]:
```suggestion
                  return <div className="error" key={key}>{errormsg}</div>;
```

File: ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rk3rn3r) [reply ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx]: Done

File: ui/packages/ui/src/app/pages/connectors/ConnectorListItem.tsx [hunk 7]
```diff
@@ -50,9 +50,9 @@ export const ConnectorListItem: React.FunctionComponent<IConnectorListItemProps>
   const getTaskStates = () => {
     const taskElements: any = [];
 
-    const stateMap = new Map(Object.entries(props.taskStates));
-    stateMap.forEach( (taskState: any, id: string) => {
-      taskElements.push(<ConnectorTask key={id} task={taskState} taskId={id} />);
+    const statesMap = new Map(Object.entries(props.taskStates));
+    statesMap.forEach( (taskState: any, id: string) => {
+      taskElements.push(<ConnectorTask key={id} task={taskState.taskStatus} taskId={id} errors={taskState.errors} />);
```
[the remark this answers is not in the document]
Comment (rk3rn3r) [reply ui/packages/ui/src/app/pages/connectors/ConnectorListItem.tsx:56]: Done. Thank you for your guidance :pray:

File: ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx (hunk 5 above)
Comment (rk3rn3r) [reply ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx:46]: Was not able to test this. How can I have the tooltip open AND examine it in dev toolbar? pseudo-classes like hover don't work. Isn't that bad UX / accessibilty? the tooltip should always be there for disabled people and only be shown on :hover ? WDYT? should we log an issue for that? @uidoyen  @indraraj

**Comment by rk3rn3r** [comment]
@gunnarmorling Delete endpoint comes in a different PR today!
I won't do the FE implementation, but that should be easy for the guys!

**Review by indraraj — APPROVED** [review]
Looks good.

File: ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx (hunk 5 above)
Comment (rk3rn3r) [reply ui/packages/ui/src/app/pages/connectors/ConnectorTask.tsx]: I logged #192  for it

**Pull request merged**